app/jitsi/main.py

```diff
@@ -0,0 +1,33 @@
+import time
+from threading import Thread
+from .database import create_user, update_stale_users
+from flask import Flask, send_from_directory, redirect, url_for, Blueprint, current_app, request, jsonify
+
+main = Blueprint('main', __name__)
+
+@main.before_app_first_request
+def before_first_request():
+    print('starting thread')
+
+    '''Start a background thread that marks stale users as offline'''
+    def find_offline_users(app):
+        with app.app_context():
+            while True:
+                update_stale_users()
+                time.sleep(10)
+
+    thread = Thread(target=find_offline_users, args=(current_app._get_current_object(),))
+    thread.start()
```

Comment:
@hndrewaall what do you think of this approach?

Reply:
I think you can do away with `user.online` and just check the value of `last_seen` at lookup time

Reply:
FYI I think the "correct" way to do this, if it turns out to be necessary, is to have a separate "db manager" application that runs completely offline, rather than a thread within this online app